metrotranscom/doorway
# 41/create etl task

# Pull Request Template

## Issue Overview

This PR addresses #issue


## Description

This PR introduces the code for the ETL task that pulls listings from the Bloom endpoint and adds them into the database table created by #42.  It does not include the Dockerfile for that task

## How Can This Be Tested/Reviewed?

To run the unit tests, call `yarn junit` or `yarn test:cov` in the package root.  Test coverage output should look something like this:

<img width="630" alt="Screenshot 2023-04-20 at 11 01 49 AM" src="https://user-images.githubusercontent.com/23032560/233440990-3e44f698-c8ef-4c61-ab91-b43b9f2d5717.png">

To run the import process, you must first checkout the `31/create-combined-listings-endpoint` branch and run `yarn db:reseed` in the `backend/core` directory to create the database table.  Then switch back to this branch, go to `tasks/bloom-etl`, and run `yarn install` and `yarn build`.  Set the DATABASE_URL env var to your local postgres instance (probably `export DATABASE_URL=postgres://bloom-dev:bloom@localhost:5432/bloom`, and run `yarn import:run`.  You should get output that looks like this:

<img width="577" alt="image" src="https://user-images.githubusercontent.com/23032560/233445043-36dfb797-aea7-41a0-a6a7-9dc25451cae7.png">

## Checklist:


## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.
